### ISSUE TITLE
'eosiolib/eosio.hpp' file not found

### ISSUE THREAD

**Issue by newbieh4cker** [EOSIO/eos#5129]
using macOS
cloned master branch from eosio repo

sudo ./eosio_build.sh -s EOS
sudo ./eosio_install.sh
cd build; sudo make install

still, 'eosiolib/eosio.hpp' file not found error occurs.
seems like it is common issue these days. Is there any solution? (except for downgrade)

**Comment by haorendashu** [comment]
seems like it is common issue these days. Is there any solution? (except for downgrade)

**Comment by lazaridiscom** [comment]
try #5133

**Comment by newbieh4cker** [comment]
@lazaridiscom 
I'm using macOS, but still same error occurs

**Comment by lazaridiscom** [comment]
clone the repo without sudo, apply the revert, build without sudo, then sudo make install.

**Comment by newbieh4cker** [comment]
@lazaridiscom 
still same error T.T

**Comment by lazaridiscom** [comment]
(sidenote: no need to ping, as a commenter I get anyway notifications)

Possibly this is a special case. You could add full version informations and the exact commands you uses (e.g. copy of your terminal).

If you just want to test etc., you can go back to an earlier version, or use docker.

**Comment by newbieh4cker** [comment]
git clone https://github.com/EOSIO/eos --recursive
cd eos
git revert -n 67393d7
./eosio_build.sh -s EOS
cd build
sudo make install

cd ~
mkdir eos-contract
cd eos-contract

eosiocpp -n hello
cd hello
eosiocpp -o hello.wast hello.cpp


hello.cpp:1:10: fatal error: 'eosiolib/eosio.hpp' file not found
#include <eosiolib/eosio.hpp>
         ^~~~~~~~~~~~~~~~~~~~
1 error generated.

**Comment by newbieh4cker** [comment]
stiiiiiiiiiil same error occurs, perhaps I have to wait
thanks a lot

**Comment by andriantolie** [comment]
Please share your `which eosiocpp` execution result :)

**Comment by lazaridiscom** [comment]
@newbieh4cker , please type those commands and paste the output:

```bash
locate eosiolib
which eosiocpp
```

**Comment by lonkly** [comment]
Hello, the same problem here, revert after re-install does not help. 

`locate eosiolib` gives 
```
/usr/local/eosio/include/eosiolib
/usr/local/eosio/include/eosiolib/CMakeLists.txt
/usr/local/eosio/include/eosiolib/action.h
/usr/local/eosio/include/eosiolib/action.hpp
/usr/local/eosio/include/eosiolib/asset.hpp
/usr/local/eosio/include/eosiolib/chain.h
/usr/local/eosio/include/eosiolib/compiler_builtins.h
/usr/local/eosio/include/eosiolib/contract.hpp
/usr/local/eosio/include/eosiolib/contracts.dox
/usr/local/eosio/include/eosiolib/core_symbol.hpp.in
/usr/local/eosio/include/eosiolib/crypto.h
/usr/local/eosio/include/eosiolib/currency.hpp
/usr/local/eosio/include/eosiolib/datastream.hpp
/usr/local/eosio/include/eosiolib/db.h
/usr/local/eosio/include/eosiolib/dispatcher.hpp
/usr/local/eosio/include/eosiolib/eosio.hpp
/usr/local/eosio/include/eosiolib/eosiolib.cpp
/usr/local/eosio/include/eosiolib/fixed_key.hpp
/usr/local/eosio/include/eosiolib/fixedpoint.hpp
/usr/local/eosio/include/eosiolib/mainpage.md
/usr/local/eosio/include/eosiolib/memory.h
/usr/local/eosio/include/eosiolib/memory.hpp
/usr/local/eosio/include/eosiolib/multi_index.hpp
/usr/local/eosio/include/eosiolib/optional.hpp
/usr/local/eosio/include/eosiolib/permission.h
/usr/local/eosio/include/eosiolib/permission.hpp
/usr/local/eosio/include/eosiolib/print.h
/usr/local/eosio/include/eosiolib/print.hpp
/usr/local/eosio/include/eosiolib/privileged.h
/usr/local/eosio/include/eosiolib/privileged.hpp
/usr/local/eosio/include/eosiolib/producer_schedule.hpp
/usr/local/eosio/include/eosiolib/public_key.hpp
/usr/local/eosio/include/eosiolib/reflect.hpp
/usr/local/eosio/include/eosiolib/rpc.dox
/usr/local/eosio/include/eosiolib/serialize.hpp
/usr/local/eosio/include/eosiolib/singleton.hpp
/usr/local/eosio/include/eosiolib/stdlib.hpp
/usr/local/eosio/include/eosiolib/symbol.hpp
/usr/local/eosio/include/eosiolib/system.h
/usr/local/eosio/include/eosiolib/time.hpp
/usr/local/eosio/include/eosiolib/transaction.h
/usr/local/eosio/include/eosiolib/transaction.hpp
/usr/local/eosio/include/eosiolib/types.h
/usr/local/eosio/include/eosiolib/types.hpp
/usr/local/eosio/include/eosiolib/varint.hpp
/usr/local/eosio/include/eosiolib/vector.hpp
/usr/local/eosio/usr/share/eosio/contractsdk/lib/eosiolib.bc
/usr/local/usr/share/eosio/contractsdk/lib/eosiolib.bc
/usr/local/eosio/bin//eosiocpp
```

`which eosiocpp`
gives 
`/usr/local/eosio/bin//eosiocpp`

**Comment by chuang39** [comment]
Still seeing the issue. Same phenomenon as people mentioned above..

**Comment by chuang39** [comment]
I can compile the contract before. After pull the latest code, got this error.

**Comment by newbieh4cker** [comment]
@lazaridiscom 

`locate eosiolib` result is same as @lonkly  mentioned.
`which eosiocpp` returns "/usr/local/bin/eosiocpp"

**Comment by lazaridiscom** [comment]
I'll take a look at all this tomorrow again. possibly I can come up with an additional patch

For anyone having still problems:

* Ensure to apply the steps within #5133 strictly.
* Ensure that the revert is applied

**Comment by lazaridiscom** [comment]
See #5133, "Further Steps" section

**Comment by lazaridiscom** [comment]
@newbieh4cker , try `rm -rf /usr/local/bin/eosiocpp`, this deletes the older eosiocpp

**Comment by newbieh4cker** [comment]
@lazaridiscom 

"further steps" did fix eosio.hpp error, but another error occurs.

![image](https://user-images.githubusercontent.com/23703741/44008769-e77f4eda-9ee1-11e8-965d-7d5412a2048d.png)

**Comment by lazaridiscom** [comment]
`locate stdint.h`, and repeat the further steps. You would start with something like this:

`export EOSIOCPP_CFLAGS='-I/usr/local/eosio/include -I/my/stdint/path'`

**Comment by qq1032246642** [comment]
#4578  in this PR, default dir change `usr/local` into `usr/local/eosio`
so should change `EOSIO_INSTALL_DIR` variable  in eosiocpp. 
```EOSIO_INSTALL_DIR=`dirname ${EOSIO_BIN_INSTALL_DIR}`/eosio```

**Comment by lazaridiscom** [comment]
try the latest 1.2.0, this should fix your issues (please close this issue if it works for you).

**Comment by lazaridiscom** [comment]
@newbieh4cker

**Comment by newbieh4cker** [comment]
@lazaridiscom 
working good~!

**Comment by lovesuper** [comment]
You need to use: 

`#include <eosio/crypto.hpp>`

**Comment by Ianfinn1994** [comment]
Trying to use the online ide to build a smart contract. Keep getting this error message. 'eosiolib/eosio.hpp' file not found. How can i install this?

**Comment by grajnikanth** [comment]
I got the same error. I installed EOSIO and EOSIO.CDT using the binaries for deploying my contracts using cleos. I unintalled EOSIO.CDT and installed the binary for EOSIO.CDT directly from the github page located at https://github.com/EOSIO/eosio.cdt. I was using Ubuntu on windows. So I used the "wget ..." command under Debian instructions to download and installed the EOSIO.CDT. The version I uninstalled was a previous version and now I got the EOSIO.CDT version 1.8.0. With this I was able to compile the smart contract successfully. I did get some warnings about "ricardian contract" but the code compiled and I could deploy it.
